### PR TITLE
feat: login with privateKey

### PR DESCRIPTION
**Pull request by ieow**
Login from embed with Private Key
require controller PR https://github.com/torusresearch/controllers/pull/106
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?
not able to login with privateKey

Issue Number: N/A

## What is the new behavior?

- able to login from embed with private key
-
-

## Other information
Require Controller PR https://github.com/torusresearch/controllers/pull/106

### CODE REVIEW

File: src/controllers/TorusController.ts [hunk 1]
```diff
@@ -22,6 +22,8 @@ import {
   FEATURES_PROVIDER_CHANGE_WINDOW,
   getPopupFeatures,
   ICommunicationProviderHandlers,
+  IEhandler,
```
Comment (himanshuchawla009) [inline src/controllers/TorusController.ts:25]: why you are using IEHandler here not using JRPCRequest type here?

File: src/controllers/TorusController.ts [hunk 2]
```diff
@@ -1354,6 +1356,13 @@ export default class TorusController extends BaseController<TorusControllerConfi
     end();
   }
 
+  private async loginWithPrivateKey(req: IEhandler<LoginWithPrivateKeyParams>): Promise<{ success: boolean }> {
+    if (!req.params?.privateKey) throw new Error("Invalid Private Key");
+    const address = await this.addAccount(req.params?.privateKey, req.params?.userInfo);
```
Comment (himanshuchawla009) [inline src/controllers/TorusController.ts:1361]: you need to toggle widget status in embed once the user is logged in (is it happening already).

File: src/controllers/TorusController.ts (hunk 1 above)
Comment (ieow) [reply src/controllers/TorusController.ts:25]: I wrapped loginWithPrivateKey provider handler with createGenericJRPCMiddleware.
createGenericJRPCMiddleware can easily plug in new provider handler but have Interface IEhandler which extends JRPCRequest with { windowId?:string , origin?:string }

File: src/controllers/TorusController.ts (hunk 2 above)
Comment (ieow) [reply src/controllers/TorusController.ts:1361]: Yes, by default the widget is visible once user is login (has keypair and selectedAddress)
unless is set hidden ( isIframeFullScreen = false ) by embed 
Current Solana embed default setting is set hidden

**Pull request merged**